src/wallet/scriptpubkeyman.h

```diff
@@ -0,0 +1,477 @@
+    bool IsWalletFlagSet(uint64_t flag) const { return storage.IsWalletFlagSet(flag); }
+    void SetWalletFlag(uint64_t flag) { return storage.SetWalletFlag(flag); }
+    void UnsetWalletFlagWithDB(WalletBatch& batch, uint64_t flag) { return storage.UnsetWalletFlagWithDB(batch, flag); }
+    bool CanSupportFeature(enum WalletFeature wf) const { return storage.CanSupportFeature(wf); }
+    void SetMinVersion(enum WalletFeature wf) { return storage.SetMinVersion(wf); }
+    const CKeyingMaterial& GetEncryptionKey() const { return storage.GetEncryptionKey(); }
+    bool HasEncryptionKeys() const { return storage.HasEncryptionKeys(); }
+    bool IsLocked() const { return storage.IsLocked(); }
+
+    virtual ~ScriptPubKeyMan() {};
+    virtual bool GetNewDestination(const OutputType type, CTxDestination& dest, std::string& error) { return false; }
+    virtual isminetype IsMine(const CScript& script) const { return ISMINE_NO; }
+
+    virtual bool HasEncryptedData() const { return false; }
+
+    virtual bool CheckDecryptionKey(const CKeyingMaterial& master_key, bool accept_no_keys = false) { return false; }
+    virtual bool Encrypt(const CKeyingMaterial& master_key, WalletBatch* batch) { return false; }
+
+    virtual bool GetReservedDestination(const OutputType type, bool internal, CTxDestination& address, int64_t& index, CKeyPool& keypool) { return false; }
+    virtual void KeepDestination(int64_t index) {}
+    virtual void ReturnDestination(int64_t index, bool internal, const CTxDestination& addr) {}
+
+    virtual bool TopUp(unsigned int size = 0) { return false; }
+
+    //! Mark unused addresses as being used
+    virtual void MarkUnusedAddresses(const CScript& script) {}
+
+    //! Upgrade stored CKeyMetadata objects to store key origin info as KeyOriginInfo
+    virtual void UpgradeKeyMetadata() {}
+
+    /** Sets up the key generation stuff, i.e. generates new HD seeds and sets them as active.
+      * Returns false if already setup or setup fails, true if setup is successful
+      * Set force=true to make it re-setup if already setup, used for upgrades
+      */
+    virtual bool SetupGeneration(bool force = false) { return false; }
+
+    /* Returns true if HD is enabled */
+    virtual bool IsHDEnabled() const { return false; }
+
+    /* Returns true if the wallet can give out new addresses. This means it has keys in the keypool or can generate new keys */
+    virtual bool CanGetAddresses(bool internal = false) { return false; }
+
+    /** Upgrades the wallet to the specified version */
+    virtual bool Upgrade(int prev_version, int new_version, std::string& error) { return false; }
+
+    virtual bool HavePrivateKeys() const { return false; }
+
+    //! The action to do when the DB needs rewrite
+    virtual void RewriteDB() {}
+
+    virtual int64_t GetOldestKeyPoolTime() { return GetTime(); }
+
+    virtual size_t KeypoolCountExternalKeys() { return 0; }
+    virtual unsigned int GetKeypoolSize() const { return 0; }
```

Comment:
Might be good to standardize on `GetKeyPoolSize` pr `GetKeypoolSize` (both are used after this change)

Reply:
Standardized on `GetKeyPoolSize` as that was already being used.